apps/src/javalab/JavalabView.jsx

```diff
@@ -107,6 +137,83 @@ class JavalabView extends React.Component {
     });
   };
 
+  handleWidthResize = desiredWidth => {
+    let newWidth = Math.max(100, Math.min(desiredWidth, 600));
+    this.props.setLeftWidth(newWidth);
+
+    this.updateLayoutThrottled(newWidth);
+  };
+
+  updateLayout = availableWidth => {
```

Comment:
There's an opportunity for some good unit tests here and in a few other places. If you decide to add testing as a follow-up, add a task.

Reply:
Good point.  Added to backlog.